src/nvim/buffer.c

```diff
@@ -4610,6 +4618,25 @@ chk_modeline (
 }
 
 /*
+ * checks the modeline is not longer
+ * than the INT_MAX 
+ */
+static bool isVersionNumberTooLong(char_u* e)
+{
+  int digitsFound = 0;
+  char myBuffer[20];
+ 
+  snprintf(&myBuffer[0], 20, "%d",INT_MAX);
+  int longestVersionPossible = strlen(myBuffer);
+ 
+  for (digitsFound = 0;
+       ascii_isdigit(*e) && digitsFound < 20;
+       digitsFound++,e++);
```

Comment:
should this have an empty block `{}` to make it explicit? I find it easy to misread empty-body loops without explicit empty blocks.